Feat login

bnert opened the pull request.

Commit:
scripts: added scripts to wrap docker cmd's

Each of the scripts abstracts away some of the steps
to pull down an sftp image, run it and connect
using a pre-defined set of username/password.

The docker image shouldn't be used in a prod setting

Commit:
feat__login: Created ClientWrapper which logs into server

The files give a class which wraps a SFTP library. The library
provides a Connect() command, which automatically uses the
credentials provided to initiate and maintain a session with the server.

In addtion:

1. Addressed comment on previous PR regarding where the 'IsConnected'
variable is set.

2. Updated the scripts to install/run the docker sftp image.

Commit:
feat__login: Created ClientWrapper which logs into server

The files give a class which wraps a SFTP library. The library
provides a Connect() command, which automatically uses the
credentials provided to initiate and maintain a session with the server.

karikittie commented:
@brent-soles It looks like this is a PR for the work I took on, so I'm gonna assign that ticket to you.

karikittie commented on FtpClient/FtpCli/Pkgs/SftpClient.cs:
```diff
@@ -0,0 +1,55 @@
+             {
+                Console.WriteLine($"Unable to connect to {host}:{port}");
+                Console.WriteLine(err.ToString());
+                Environment.Exit(-1);
```
Do we want to exit here? It might be better to handle the program execution logic as near to the main method as possible so we don't clutter the execution with arbitrary exit codes around the program? Could we just throw this error?

bnert replied:
We could. My main reason for putting this there was I didn't see the need for further execution if the client is unable to connect.

If this was to be made throwing an exception, we (the team in general) should figure out how deep we want to nest try/catch blocks. I think having them been in the Logic Layer would make the most sense, but again we can all discuss it.

karikittie commented on FtpClient/FtpCli/Pkgs/SftpClient.cs:
```diff
@@ -0,0 +1,55 @@
+
+
+        public bool IsConnected {
+          get { return _isConnected; }
```
Is there any case where this returns false?

bnert replied:
It can. When it is time to implement timing out from the server, this can be used in the main event loop to exit upon disconnecting. I realize I should change this to return the library

The library (SSH.NET) Connect() method can be invoked with a Timeout, which will be useful.

karikittie commented on FtpClient/FtpCli/Program.cs:
```diff
@@ -2,11 +2,59 @@
+            // if an incorect number of parameters were passed
+            // print usage details and prompt user again
+            string usage = "servername username [cmd, args, ...]";
+            while (args.Length < 2) {
```
This is not the way the application is working according to the other open PR right now: https://github.com/cs410agile-group3/FTPClient/pull/50

bnert replied:
I branched this from Michelle's earlier work before she refactored. Should be fixed on merge.

karikittie reviewed (commented):
A few comments

Commit:
Merge branch 'feat_login' of https://github.com/cs410agile-group3/FTPClient into feat_login

This commit fixes some conflicts when pushing up changes to feat_login.

Commit:
feat_login: cleaned up code and reorganized naming

In addtion, This commit resolves some extra files being added
due to a rebase to fix some commit conflicts.

mkduer reviewed (approved):
Requested a `README.md` update regarding the docker script and container usage. Spoke offline to confirm that an exception will be thrown in `SftpClient.cs` if a connection isn't made in the `void Connect()` method and Brent confirmed that this is also shown through testing. Other than that, code looks good.

Commit:
docs/bash: updated documentation & script bugs

Commit:
Merge branch 'develop' into feat_login

Commit:
feat_login: fixed system exit

Updated the code to throw an exception rather than
exit when the client is unable to connect to the
server.

Commit:
Merge branch 'feat_login' of https://github.com/cs410agile-group3/FTPClient into feat_login
